### ISSUE TITLE
Generic type T extends `{ domain:string }`

### ISSUE THREAD

**Issue by kraenhansen** [psrebniak/swagger-typescript-client-generator#1]
It's unclear to me why the client class [needs a configuration which extends](https://github.com/psrebniak/swagger-typescript-client-generator/blob/207fef39d3371c112225afbfc6a86a2fc958c7ad/src/typescriptConverter.ts#L219)

```typescript
{ domain:string }
```

I think it would be more intuitive if the client config was just extending an object.

**Comment by psrebniak** [comment]
Hi @kraenhansen 
initial idea was to add some request factory types and use domain as a required parameter for every factory type. 
Do you have some use cases where client should work without domain/host?

**Comment by kraenhansen** [comment]
The way I am using your package is by generating the types and client class (saved to `./generated.ts`) and the extending that to expose the client. This is very much work in progress, but perhaps it gives a better picture of what I'm doing:

```typescript
import fetch from "isomorphic-fetch";
import qs from "querystring";

import * as generated from "./generated";

interface NAGConfiguration {
  authorization?: string;
  hostname?: string;
  clientId: string;
  clientSecret: string;
}

export class NAGClient extends generated.NAGClient<
  NAGConfiguration & { domain: string }
> {
  static DEFAULT_BASE_URL = "https://api.nordicapigateway.com/v1/";

  static requestFactory: generated.RequestFactoryType = (
    path: string,
    query: any,
    body: any,
    formData: any,
    headers: Headers,
    method: string,
    configuration: NAGConfiguration
  ) => {
    const {
      authorization,
      clientId,
      clientSecret,
      hostname = NAGClient.DEFAULT_BASE_URL
    } = configuration;

    const url = new URL(path, hostname);
    url.search = qs.encode(query);

    const parsedHeaders = new Headers(headers);
    parsedHeaders.set("X-Client-ID", clientId);
    parsedHeaders.set("X-Client-Secret", clientSecret);
    parsedHeaders.set("Authorization", authorization);

    if (formData) {
      throw new Error("This client doesn't support form data");
    }

    return fetch(url.href, { headers, body, method }).then(res => {
      const contentType = res.headers.get("Content-Type");
      if (contentType === "application/json") {
        return res.json();
      } else {
        throw new Error(
          `Expected a JSON response from the API, got ${contentType}`
        );
      }
    });
  };

  private config: NAGConfiguration;

  constructor(config: NAGConfiguration) {
    super({ domain: "", ...config }, NAGClient.requestFactory);
    this.config = config;
  }

  public withAuthorization(authorization: string) {
    return new NAGClient({
      ...this.config,
      authorization
    });
  }
}

```

In this case the domain (or hostname) will be an optional configuration parameter: I am building a client for a specific API where the hostname could be considered a part of the agreed protocol.

**Comment by psrebniak** [comment]
@kraenhansen feel free to create PR for fixing that issue

**Comment by psrebniak** [comment]
@kraenhansen fixed in v1.2.0

**Comment by kraenhansen** [comment]
Great - thanks!
